ciphers/mixed_keyword_cypher.py

```diff
@@ -19,50 +21,49 @@ def mixed_keyword(key: str = "college", pt: str = "UNIVERSITY") -> str:
      'Y': 'T', 'Z': 'Y'}
     'XKJGUFMJST'
     """
-    key = key.upper()
-    pt = pt.upper()
-    temp = []
-    for i in key:
-        if i not in temp:
-            temp.append(i)
-    len_temp = len(temp)
-    # print(temp)
-    alpha = []
-    modalpha = []
-    for j in range(65, 91):
-        t = chr(j)
-        alpha.append(t)
-        if t not in temp:
-            temp.append(t)
-    # print(temp)
-    r = int(26 / 4)
-    # print(r)
+    keyword = keyword.upper()
+    plaintext = plaintext.upper()
+
+    unique_chars = []
+    for char in keyword:
+        if char not in unique_chars:
+            unique_chars.append(char)
+
+    num_unique_chars_in_keyword = len(unique_chars)
+
+    alphabet = list(ascii_uppercase)
+    # add the rest of the alphabet to the unique_chars list
+    for char in alphabet:
+        if char not in unique_chars:
+            unique_chars.append(char)
+
+    rows = int(26 / 4)
+    # k is an index variable
     k = 0
-    for _ in range(r):
-        s = []
-        for _ in range(len_temp):
-            s.append(temp[k])
+    modified_alphabet = []
```

Comment:
Please do all this with `set`

Reply:
As this is a basic substitution cypher, it is bounded by the number of different letters in the alphabet we're using. Using sets in this case wouldn't change the performance almost at all.